server/controllers/ArticleController.js

```diff
@@ -0,0 +1,252 @@
+          });
+        }
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method gets all articles from the database,
+   * It returns a json response of the article retrieved
+   * It handles the GET /api/articles.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getAllArticle(req, res, next) {
+    Article.findAll({
+      include:
+      [{ model: User }],
+      order: [['createdAt', 'DESC']],
+    })
+      .then((articles) => {
+        if (articles.length > 0) {
+          const articlesList = articles.map(article => ({
+            id: article.id,
+            title: article.title,
+            description: article.description,
+            body: article.body,
+            slug: article.slug,
+            image: article.image,
+            createdAt: article.createdAt,
+            updatedAt: article.updatedAt,
+            author: {
+              id: article.User.id,
+              username: article.User.username,
+            },
+          }));
+
+          res.status(200).json({
+            articles: articlesList,
+          });
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'No Articles found',
+          });
+        }
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method updates an article in the database,
+   * It returns a json response of the article updated
+   * It handles the PUT /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static updateArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          const updatedArticle = article;
+          updatedArticle.title = title !== undefined ? title : article.title;
+          updatedArticle.description = description
+          !== undefined ? description : article.description;
+          updatedArticle.body = body !== undefined ? body : article.body;
+          updatedArticle.save();
+          User.find({ where: { id: article.authorId } })
+            .then((user) => {
+              res.status(200).json({
+                article: {
+                  id: article.id,
+                  title: article.title,
+                  description: article.description,
+                  body: article.body,
+                  slug: article.slug,
+                  image: article.image,
+                  createdAt: article.createdAt,
+                  updatedAt: article.updatedAt,
+                  author: {
+                    id: user.id,
+                    username: user.username,
+                  },
+                },
```

Comment:
I think there is a more efficient way to do an `update` in sequelize. Over here you are doing 3 queries but I think you can make it just one query. This link might help you figure our how to achieve this https://stackoverflow.com/questions/8158244/how-to-update-a-record-using-sequelize-for-node/28986988

Reply:
done